internal/pkg/client/progress/progress_roundtripper.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package progress
+
+import (
+	"net/http"
+)
+
+const contentSizeThreshold = 1024
+
+type RoundTripper struct {
+	inner http.RoundTripper
+	pb    *DownloadBar
+}
+
+func NewRoundTripper(inner http.RoundTripper, pb *DownloadBar) *RoundTripper {
+	if inner == nil {
+		inner = http.DefaultTransport
+	}
+
+	rt := RoundTripper{
+		inner: inner,
+		pb:    pb,
+	}
+
+	return &rt
+}
+
+func (t *RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if t.pb != nil && req.Body != nil && req.ContentLength >= contentSizeThreshold {
+		t.pb.Init(req.ContentLength)
```

Comment:
Ideally every `Init` is paired with an `Abort` or `Wait` call when finished. A bit awkward here.... but if there's a way, then it should be added.

`mpb` implements multi-progress bar containers. It's expected that additional bars might be added, so the go routines associated with mpb don't exit until they are told to.